micro-rdk/src/common/data_manager.rs

```diff
@@ -70,7 +77,7 @@ fn get_data_sync_interval(cfg: &ConfigResponse) -> Result<Option<Duration>, Data
 
 pub struct DataManager<StoreType> {
     collectors: Vec<DataCollector>,
-    store: StoreType,
+    store: Rc<AsyncMutex<StoreType>>,
```

Comment:
You don't have multiple readers so an rwlock doesn't gain you anything. But you could potentially use `BiLock` since you have exactly one reader and one writer.  I can't say it'll definitely work because I haven't used `BiLock` but it seems to fit the description.

Reply:
I agree, I will attempt to trylock bilock

Reply:
Actually it requires a separate feature to be passed to futures-util, should I still proceed?

Reply:
No, Mutex is fine then I think.